contracts/lockdrop/src/contract.rs

```diff
@@ -1848,6 +2056,37 @@ fn calculate_weight(amount: Uint128, duration: u64, config: &Config) -> Uint256
     lock_weight * amount.into()
 }
 
+/// ## Description
+/// Calculates bLuna user reward according to his share in LP  
+fn calc_user_reward(
+    deps: DepsMut,
+    user: &Addr,
+    user_lp_amount: Uint128,
+    total_lp_amount: Uint128,
+    total_reward_balance: Uint256,
+) -> StdResult<Uint128> {
+    if user_lp_amount.is_zero() || total_lp_amount.is_zero() {
+        return Ok(Uint128::zero());
+    }
+
+    let to_distribute = match USERS_BLUNA_REWARD_INDEX.may_load(deps.storage, user)? {
+        None => total_reward_balance,
+        Some(last_user_bluna_reward_index)
+            if last_user_bluna_reward_index < total_reward_balance =>
+        {
+            total_reward_balance - last_user_bluna_reward_index
+        }
+        _ => Uint256::zero(),
```

Comment:
Here, we can return Uint256:zero() and skip steps below.

Reply:
Sure, thank you.